RiotSwiftUI/Modules/Template/SimpleUserProfileExample/Model/TemplateUserProfileViewState.swift

```diff
@@ -0,0 +1,23 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+struct TemplateUserProfileViewState {
+    let avatar: AvatarInputType?
+    let displayName: String?
+    var presence: TemplateUserProfilePresence = .offline
```

Comment:
Having immutable structure with immutable properties is safer to guarantee functional programming principles. We can set the default value like this otherwise:
```
init(avatar: AvatarInputType?, displayName, presence = .offline)
```